t/029-tool-application-support.t.cpp

```diff
@@ -17,112 +17,127 @@
-      llvm::install_fatal_error_handler(&fatalErrorHandler, 0);
-    }
+      std::vector<const char*> argumentPtrs;
+      for (const auto& item : arguments) {
+        argumentPtrs.push_back(item.c_str());
+      }
 
-    ~ToolApplicationSupportTest()
-    {
-      llvm::remove_fatal_error_handler();
-    }
+      int argc = argumentPtrs.size();
+      const char** argv = argumentPtrs.data();
+      clang::tooling::CommonOptionsParser parser(argc, argv,
+                                                 optionCategory());
 
-    std::string verifyInstallation(const std::vector<std::string>& arguments)
-    {
-      try {
-        std::vector<const char*> argumentPtrs;
-        for (const auto& item : arguments) {
-          argumentPtrs.push_back(item.c_str());
-        }
-
-        int argc = argumentPtrs.size();
-        const char** argv = argumentPtrs.data();
-        clang::tooling::CommonOptionsParser parser(argc, argv,
-                                                   optionCategory());
-
-        clangmetatool::ToolApplicationSupport::verifyInstallation(
-          parser.getCompilations(),
-          parser.getSourcePathList());
-      }
-      catch(const LLVMFatalError& e) {
-        return e.what();
-      }
+      // ToolApplicationSupport::verifyInstallation crashes with exit(1)
+      // if the required headers aren't found
+      clangmetatool::ToolApplicationSupport::verifyInstallation(
+        parser.getCompilations(),
+        parser.getSourcePathList());
 
-      return std::string();
+      return true;
     }
 };
 
-TEST_F(ToolApplicationSupportTest, resource_dir)
+TEST_F(ToolApplicationSupportTest, ARCH_DEPENDENT(ResourceDirFound))
 {
-  EXPECT_EQ("", verifyInstallation({"tool", "test.cpp", "--",
-                                    "-resource-dir", CLANG_STD_INCLUDE_DIR}));
+  EXPECT_TRUE(verifyInstallation({"tool", "test.cpp", "--",
+                                 "-resource-dir", CLANG_STD_INCLUDE_DIR}))
+    << "verification failed with CLANG_STD_INCLUDE_DIR=" CLANG_STD_INCLUDE_DIR
+    << "\n";
+}
 
-  EXPECT_EQ("clang resource files are missing from /non-existent/123, "
-            "check that this application is installed properly",
-            verifyInstallation({"tool", "test.cpp", "--",
-                                "-resource-dir", "/non-existent/123"}));
+TEST_F(ToolApplicationSupportTest, ARCH_DEPENDENT(ResourceDirNotFound))
+{
+  const char *messageRegex = ".*"
+                             "clang resource files are missing from "
+                             "/non-existent/123, check that this application "
+                             "is installed properly";
+  ASSERT_DEATH(
+    verifyInstallation({"tool", "test.cpp", "--",
+                        "-resource-dir", "/non-existent/123"}),
+    messageRegex);
+}
+
+TEST_F(ToolApplicationSupportTest, ARCH_DEPENDENT(GCCToolChain))
+{
+  const char *messageRegex = ".*"
+                             "clang could not find a C++ toolchain to use, "
```

Comment:
I think the tests may be failing because you need to escape the `+` here

Reply:
That was it. Fixed that and the Dockerfile